src/mmrm.cpp

```diff
@@ -52,30 +53,39 @@ Type objective_function<Type>::operator() ()
   vector<Type> y_vec_tilde = vector<Type>::Zero(y_vector.rows());
   // Sum of the log determinant will be incrementally calculated here.
   Type sum_log_det = 0.0;
-
   // Create the lower triangular Cholesky factor of the visit x visit covariance matrix.
-  matrix<Type> covariance_lower_chol = get_covariance_lower_chol<Type>(theta, n_visits, cov_type);
-
+  matrix<Type> covariance_lower_chol = matrix<Type>::Zero(n_visits, n_visits * n_groups);
+  int covariance_size = theta.size() / n_groups;
+  for (int i = 0; i < n_groups; i++) {
+    matrix<Type> lower_chol = get_covariance_lower_chol<Type>(theta.segment(i * n_groups, covariance_size), n_visits, cov_type);
+    for (int j = 0; j < n_visits; j++) {
+      covariance_lower_chol.col(j + i * n_visits) = lower_chol.col(j);
+    }
+  }
```

Comment:
since now this is more complex, please move to a function e.g.   `get_cov_lower_chol_grouped<Type>(theta, n_visits, cov_type, n_groups)` so that you can unit test that